11-algorithms.Rmd

```diff
@@ -36,27 +37,35 @@ The example also reflects a secondary aim of the chapter which, following @xiao_
 
 ## Scripts
 
-If functions distributed in packages are the building blocks of R code, scripts are the glue that holds them together, in a logical order, to create reproducible workflows.
-To programming novices scripts may sound intimidating but they are simply plain text files, typically saved with an extension representing the language they contain.
-R scripts are generally saved with a `.R` extension and named to reflect what they do.
-An example is `11-hello.R`, a script file stored in the `code` folder of the book's repository, which contains the following two lines of code:
+If functions distributed in packages are the building blocks of R code, scripts are the glue that holds them together.
+Scripts should be stored and executed in a logical order to create reproducible workflows, manually or with workflow automation tools such as **targets** [@landau_targets_2021].
+If you are new to programming scripts may seem intimidating when you first encounter them, but they are simply plain text files.
+Scripts are usually saved as a file with an extension representing the language they contain, such as `.py` for scripts written in Python or `.rs` for scripts written in Rust.
+R scripts should be saved with a `.R` extension and named to reflect what they do.
+An example is [`11-hello.R`](https://github.com/geocompx/geocompr/blob/main/code/11-hello.R), a script file stored in the [`code`](https://github.com/geocompx/geocompr/blob/main/code/) folder of the book's repository.
+`11-hello.R` is a simple script containing only two lines of code, one of which is a comment:
 
 ```r
 # Aim: provide a minimal R script
 print("Hello geocompr")
 ```
 
-The lines of code may not be particularly exciting but they demonstrate the point: scripts do not need to be complicated.
-Saved scripts can be called and executed in their entirety with `source()`, as demonstrated below which shows how the comment is ignored but the instruction is executed:
+The contents of this script are not particularly exciting but they demonstrate the point: scripts do not need to be complicated.
+Saved scripts can be called and executed in their entirety from the R command line with the `source()` function, as demonstrated below.
+The output of this command shows that the comment is ignored but `print()` command is executed:
 
 ```{r 10-algorithms-1}
 source("code/11-hello.R")
 ```
 
-There are no strict rules on what can and cannot go into script files and nothing to prevent you from saving broken, non-reproducible code.^[
-Lines of code that do not contain valid R should be commented out, by adding a `#` to the start of the line, to prevent errors.
-See line 1 of the `11-hello.R` script.
-]
+You can also call R scripts from system command line shells such as `bash` and `PowerShell`, for example as follows:
```

Comment:
assuming that Rscript is in your PATH variable -:) which is mostly the case for Unix systems by default but not necessarily for Windows

Reply:
Job done: You can also call R scripts from system command line shells such as `bash` and `PowerShell`, assuming that the `RScript` executable is [configured](https://www.reddit.com/r/Rlanguage/comments/zaovly/is_anybody_able_to_run_a_r_script_in_powershell/) so it is available, for example as follows: